rust/private/rustc.bzl

```diff
@@ -1304,17 +1448,34 @@ def _get_dir_names(files):
         dirs[f.dirname] = None
     return dirs.keys()
 
-def add_crate_link_flags(args, dep_info, force_all_deps_direct = False):
+def add_crate_link_flags(args, dep_info, force_all_deps_direct = False, link_to_metadata = False):
```

Comment:
I believe this function is not named properly, we add these flags when compiling `rlib`s too, not just for linking actions. While renaming the function itself does not need to be in scope for this PR, could we remove the `link` word from the new parameter? `is_pipelined_compilation`, `use_metadata` come to mind.

Reply:
Done.